pkg/clients/compute/aks.go

```diff
@@ -280,38 +129,40 @@ func newManagedCluster(c *v1alpha3.AKSCluster, appID, secret string) containerse
 				{
 					Name:   to.StringPtr(AgentPoolProfileName),
 					Count:  &nodeCount,
-					VMSize: containerservice.VMSizeTypes(c.Spec.NodeVMSize),
+					VMSize: to.StringPtr(c.Spec.NodeVMSize),
+					Mode:   containerservice.AgentPoolModeSystem,
 				},
 			},
-			ServicePrincipalProfile: &containerservice.ManagedClusterServicePrincipalProfile{
-				ClientID: to.StringPtr(appID),
-				Secret:   to.StringPtr(secret),
-			},
 			EnableRBAC: to.BoolPtr(!c.Spec.DisableRBAC),
 		},
+		Identity: &containerservice.ManagedClusterIdentity{},
+	}
+	switch containerservice.ResourceIdentityType(c.Spec.Identity.Type) { //nolint:exhaustive
+	case containerservice.ResourceIdentityTypeSystemAssigned:
+		p.Identity.Type = containerservice.ResourceIdentityTypeSystemAssigned
+	case containerservice.ResourceIdentityTypeUserAssigned:
+		p.Identity.Type = containerservice.ResourceIdentityTypeUserAssigned
+		if len(c.Spec.Identity.IdentityNames) == 0 {
+			return p, errors.New(errInvalidUserAssignedManagedIdentity)
+		}
+		p.Identity.UserAssignedIdentities = make(map[string]*containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue, len(c.Spec.Identity.IdentityNames))
+		for _, n := range c.Spec.Identity.IdentityNames {
+			resourceID := fmt.Sprintf(fmtUserAssignedManagedIdentityID, subscriptionID, c.Spec.ResourceGroupName, n)
+			p.Identity.UserAssignedIdentities[resourceID] = &containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue{}
+		}
```

Comment:
I still wonder what is the use case of having multiple user identities here :) I'll dive in later.